### PR TITLE
Implement `Struct.toSlice()`

### LINKED ISSUE #474: Add `.toSlice()` extension function for Structs and Messages
One can do `Struct.toCell().asSlice()` already, but for consistency reasons we probably should have `.toSlice()` too. For example, `StringBuilder` has it, and it would also mirror the `.fromSlice()` method nicely. 

As mentioned in: https://github.com/tact-lang/tact-docs/pull/271#discussion_r1651512617

### PR DESCRIPTION
**Pull request by Gusarich**
Closes #474 

- [x] I have updated CHANGELOG.md
- [ ] I have documented my contribution in Tact Docs: https://github.com/tact-lang/tact-docs/pull/PR-NUMBER
- [x] I have added tests to demonstrate the contribution is correctly implemented: this usually includes both positive and negative tests, showing the happy path(s) and featuring intentionally broken cases
- [x] I have run all the tests locally and no test failure was reported
- [x] I have run the linter, formatter and spellchecker
- [x] I did not do unrelated and/or undiscussed refactorings

### CODE REVIEW

File: src/abi/struct.ts [hunk 1]
```diff
@@ -103,6 +103,51 @@ export const StructFunctions: Map<string, AbiFunction> = new Map([
...
+                if (args.length !== 1) {
+                    throwCompilationError(
+                        "toSlice() expects no arguments",
+                        ref,
+                    );
+                }
+                const arg = args[0]!;
+                if (arg.kind !== "ref") {
+                    throwCompilationError(
+                        "toSlice() is implemented only a struct type",
```
Comment (anton-trunov) [inline src/abi/struct.ts]: We also have `toSlice()` for string builders: https://github.com/tact-lang/tact/blob/5bec4864deccf8a3ee694914f3be0a815818488e/stdlib/std/text.tact#L30

Comment (Gusarich) [reply src/abi/struct.ts]: Then should I also change the `toCell() is implemented only a struct type` error message for `toCell()` as there's also that function in stdlib for `StringBuilder`?

Comment (anton-trunov) [reply src/abi/struct.ts]: Sure, let's add it here too

**Comment by anton-trunov** [comment]
Let's update this and merge

File: src/abi/struct.ts (hunk 1 above)
Comment (Gusarich) [reply src/abi/struct.ts]: Maybe it'll be better to change these error messages to something like `toSlice() is not implemented for type 'type'`?

Comment (anton-trunov) [reply src/abi/struct.ts]: Sure, sounds good

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**